types/matcher.d.ts

```diff
@@ -20,9 +20,14 @@ export type Match<T extends DefaultParams = DefaultParams> =
 
 export type MatcherFn = (pattern: Path, path: Path) => Match;
 
+export type MakeRegexpFn = {
+  keys: Array<{ name: string | number }>;
+  regexp: RegExp;
+};
+
 export default function makeMatcher(
   makeRegexpFn?: (
     pattern: string,
     keys?: Array<{ name: string | number }>
```

Comment:
The 2nd argument is obsolete, we can safely remove it. `keys` are returned instead.